.github/workflows/smoke-test.yaml

```diff
@@ -0,0 +1,17 @@
+name: Action Smoke Tests
+on: workflow_dispatch
+jobs:
+  run-action:
+    name: Run GH Action
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        name: Checkout Repository
+      - uses: ./
```

Comment:
I'm not really sure how this works. How can you trigger the version from the branch? From the GitHub Actions Tab, you get to a screen from where you can trigger the workflow_dispatch event manually but I think there is no way to choose a branch. So if you are not specifying the branch here, then how will the action code from the branch will be picked?

Reply:
There's actually a dropdown which lets you pick the branch exercised on the workflow, see this screencap from my fork:
![image](https://user-images.githubusercontent.com/18124107/212012603-db263024-f6d4-40d7-8f6b-ea4e1e2ae654.png)

Reply:
I wonder if that uses the branch for picking just the workflow or the whole codebase. I mean, it should pick the whole thing, right?